contracts/FungibleTokenSwitchboard.cdc

```diff
@@ -0,0 +1,143 @@
+import FungibleToken from "./FungibleToken.cdc"
+
+/// FungibleTokenSwitchboard
+///
+/// The contract that allows an account to receive payments in multiple fungible
+/// tokens using a single {FungibleToken.Receiver}
+///
+pub contract FungibleTokenSwitchboard {
+    
+    // Storage and Public Paths
+    pub let StoragePath: StoragePath
+    pub let PublicPath: PublicPath
+    pub let ReceiverPublicPath: PublicPath
+    
+    /// SwitchboardInitialized
+    ///
+    /// The event that is emited when a new Switchboard resource is created
+    ///
+    pub event SwitchboardInitialized(switchboardResourceID: UInt64)
+
+    /// VaultCapacityAdded
+    ///
+    /// The event that is emitted when a new vault capacity is added to a
+    /// switchboard resource
+    ///
+    pub event VaultCapabilityAdded(type: Type)
+
+    /// VaultCapacityRemoved
+    ///
+    /// The event that is emitted vault capacity is added removed from a 
+    /// switchboard resource
+    ///
+    pub event VaultCapabilityRemoved(type: Type)
+    
+    /// SwitchboardPublic
+    ///
+    /// The interface that enforces the method to allow anyone to check on the
+    /// available capabilities of a switchboard resource and also exposes the 
+    /// deposit method to deposit funds on it
+    ///
+    pub resource interface SwitchboardPublic {
+        pub fun getVaultCapabilities(): {Type: Capability<&{FungibleToken.Receiver}>}
+        pub fun deposit(from: @FungibleToken.Vault)
```

Comment:
I would either:

- `remove` deposit from this interface
- or make `SwitchboardPublic` to implement `FungibleToken.Receiver`

Reply:
yeah, I think that `SwitchboardPublic` should implement the receiver interface. Good catch!

Reply:
I'm afraid that's not possible [yet](https://stackoverflow.com/questions/72408565/reference-types-in-cadence)

Reply:
oh yeah. good point. I think it is fine to leave it the way it is